nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -720,7 +720,21 @@ public TransferResult<Nullness, NullnessStore> visitFieldAccess(
     Symbol symbol = ASTHelpers.getSymbol(fieldAccessNode.getTree());
     setReceiverNonnull(updates, fieldAccessNode.getReceiver(), symbol);
     Nullness nullness = NULLABLE;
-    if (!NullabilityUtil.mayBeNullFieldFromType(symbol, config, getCodeAnnotationInfo(state))) {
+    boolean fieldMayBeNull =
+        NullabilityUtil.mayBeNullFieldFromType(symbol, config, getCodeAnnotationInfo(state));
+    if (handler
+        .onDataflowVisitFieldAccess(
+            fieldAccessNode,
+            symbol,
+            state.getTypes(),
+            state.context,
+            apContext,
+            values(input),
+            updates)
+        .equals(NullnessHint.HINT_NULLABLE)) {
+      fieldMayBeNull = true;
+    }
```

Comment:
Here could we instead write (roughly) `fieldMayBeNull = NullabilityUtil.mayBeNullFieldFromType(...) || handler.onDataflowVisitFieldAccess(...).equals(HINT_NULLABLE)`?  To me this is slightly more readable, and we'll get a short-circuiting optimization from `||` to avoid the second call sometimes